File: daily/Arrays/202-Happy-Number.go
package daily

func isHappy(n int) bool {
	seen := make(map[int]bool)
	for n != 1 {
		if seen[n] {
			return false
		}
		seen[n] = true

		next_sum := 0
		for n > 0 {
			digits := n % 10
			next_sum += digits * digits
			n = n / 10
		}
		n = next_sum
	}
	return n == 1
}

File: daily/Arrays/202-Happy-Number.py
class Solution:
    def isHappy(self, n: int) -> bool:
        seen = set()
        while n != 1 and n not in seen:
            seen.add(n)

            next_sum = 0
            while n > 0:
                digit = n % 10
                next_sum += digit ** 2
                n = n // 10

            n = next_sum 
        return n == 1